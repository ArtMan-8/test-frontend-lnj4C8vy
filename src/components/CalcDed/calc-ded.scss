.calc-ded {
  position: relative;

  flex-grow: 1;
  width: 100%;
  margin: 0 0 ;
  padding: 0 0 55px;

  border: none;

  @media(min-width: 768px) {
    flex-grow: unset;
  }
}

.calc-ded__calc {
  padding: 0;

  font-weight: 500;
  font-size: 14px;
  color: var(--color-red);

  background-color: transparent;
  border: none;
  cursor: pointer;

  &:hover {
    color: var(--color-orange);
  }

  &:active {
    color: var(--color-red);
  }
}

.calc-ded__total {
  margin-top: 16px;
}

.calc-ded__total-header {
  margin: 0;

  font-weight: 500;
  font-size: 14px;
  line-height: 24px;
}

.calc-ded__list {
  margin: 0;
  padding: 0;

  list-style: none;
}

.calc-ded__row {
  padding: 18px 0;

  border-bottom: 1px solid var(--color-grey);
}

.calc-ded__label {
  position: relative;

  padding-left: 30px;

  font-weight: 500;
  font-size: 14px;

  cursor: pointer;

  span {
    color: var(--color-grey-dark);
  }

  &::before {
    content: "";
    position: absolute;
    top: -3px;
    left: -1px;

    width: 20px;
    height: 20px;

    border: 1px solid var(--color-grey);
    border-radius: 6px;
  }

  &::after {
    content: "";
    position: absolute;
    top: 1px;
    left: 3px;

    width: 11px;
    height: 6px;

    border-bottom: 2px solid transparent;
    border-left: 2px solid transparent;
    transform: rotate(-45deg);
  }
}

.calc-ded__check:checked + .calc-ded__label {
  &::before {
    background: var(--gradient-red);
    border: none;
  }

  &::after {
    border-color: var(--color-white);
  }
}

.calc-ded__check:checked:disabled + .calc-ded__label {
  &::before {
    background: var(--color-grey);
    border: none;
  }

  &::after {
    border-color: var(--color-white);
  }
}

.calc-ded__controls {
  margin-top: 28px;
  margin-bottom: 40px;

  @media(min-width: 768px) {
    display: flex;
    margin-top: 28px;
    margin-right: 50px;
    margin-bottom: 50px;
  }
}

.calc-ded__controls-header {
  margin-bottom: 30px;

  font-weight: 500;
  font-size: 14px;

  @media(min-width: 768px) {
    margin-top: 7px;
    margin-right: auto;
    margin-bottom: unset;
  }
}

.calc-ded__controls-btn {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  margin-right: 8px;
  padding: 9px 12px;

  font-size: 14px;
  color: var(--color-black);

  background: var(--color-grey-btn);
  border: none;
  border-radius: 50px;
  cursor: pointer;

  @media(min-width: 768px) {
    margin-right: unset;
    margin-left: 18px;
  }

  &:hover {
    color: var(--color-white);

    background: var(--color-red);
  }

  &:active {
    color: var(--color-white);

    background: var(--color-red);
  }

  &:disabled {
    color: var(--color-white);

    background: var(--color-grey);
  }

  &--active {
    color: var(--color-white);

    background: var(--gradient-red);;
  }
}

.calc-ded__btn-add {
  position: absolute;
  bottom: 15px;

  display: flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  margin-top: auto;
  padding: 12px;

  font-weight: 500;
  font-size: 12px;
  color: var(--color-white);

  background: var(--gradient-red);
  border: none;
  border-radius: 6px;
  box-shadow: 0 0 24px rgba(234, 0, 41, 0.33);
  cursor: pointer;

  @media(min-width: 768px) {
    padding: 18px;

    font-size: 16px;
  }

  &:hover {
    color: var(--color-white);

    background: var(--color-red);
  }

  &:active {
    color: var(--color-white);

    background: var(--color-red);
  }

  &:disabled {
    color: var(--color-white);

    background: var(--color-grey);
  }
}
