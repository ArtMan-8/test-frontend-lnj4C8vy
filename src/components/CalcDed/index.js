export { default } from './CalcDed.jsx';
