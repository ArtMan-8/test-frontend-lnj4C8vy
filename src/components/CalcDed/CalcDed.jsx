import React, { useState, useEffect } from 'react';
import T from 'prop-types';
import './calc-ded.scss';
import salaryFormat from '../../utils/salaryFormat';
import endingsYear from '../../utils/endingsYear';
import prefix from '../../utils/prefix';

const MONTH_COUNT = 12;
const TAX = 0.13;
const MAX_DEDUCTION = 260000;

const CalcDed = ({ salary }) => {
  const [isCalc, setIsCalc] = useState(false);

  useEffect(() => {
    setIsCalc(false);
  }, [salary]);

  const deductions = () => {
    const yearSalary = salary * MONTH_COUNT;
    const annualDeduction = Math.floor(yearSalary * TAX);

    const maxDeduction = Math.floor(MAX_DEDUCTION / annualDeduction);
    const restDeduction = MAX_DEDUCTION % annualDeduction;

    const allDeductions = new Array(maxDeduction).fill(annualDeduction);
    return ([...allDeductions, restDeduction]);
  };

  return (
    <fieldset className="calc-ded">
      <button
        type="button"
        className="calc-ded__calc"
        onClick={() => salary && setIsCalc(true)}
      >Рассчитать</button>

      {isCalc
      && (<div className="calc-ded__total">
        <h3 className="calc-ded__total-header">
          Итого можете внести в&nbsp;качесте&nbsp;досрочных:
        </h3>
        <ul className="calc-ded__list">
          {deductions().map((row, index) => (
            <li key={index + 1} className="calc-ded__row">
            <input
              type="checkbox"
              className="calc-ded__check visually-hidden"
              name={`tax-${index + 1}-year`}
              id={`tax-${index + 1}-year`}
            />
            <label
              htmlFor={`tax-${index + 1}-year`}
              className="calc-ded__label"
          >{salaryFormat(row)} рублей{' '}
            <span>{prefix(index + 1)} {index + 1}-{endingsYear(index + 1)} год</span></label>
          </li>
          ))}
        </ul>
      </div>)}

      <div className="calc-ded__controls">
        <h3 className="calc-ded__controls-header">Что уменьшаем?</h3>
        <button
          type="button"
          className="calc-ded__controls-btn calc-ded__controls-btn--active"
        >Платёж</button>

        <button
          type="button"
          className="calc-ded__controls-btn"
        >Срок</button>
      </div>

      <button
        type="submit"
        className="calc-ded__btn-add"
        onClick={(evt) => evt.preventDefault()}
      >Добавить</button>
    </fieldset>
  );
};

CalcDed.propTypes = {
  salary: T.any,
};

export default CalcDed;
