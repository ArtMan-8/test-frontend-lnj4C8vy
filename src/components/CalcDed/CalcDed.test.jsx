import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CalcDed from './CalcDed.jsx';

afterEach(cleanup);

test('App snapshot', () => {
  const { asFragment } = render(<CalcDed />);
  expect(asFragment()).toMatchSnapshot();
});
