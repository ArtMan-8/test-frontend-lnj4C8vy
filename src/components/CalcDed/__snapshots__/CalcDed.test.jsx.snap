// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<DocumentFragment>
  <fieldset
    class="calc-ded"
  >
    <button
      class="calc-ded__calc"
      type="button"
    >
      Рассчитать
    </button>
    <div
      class="calc-ded__controls"
    >
      <h3
        class="calc-ded__controls-header"
      >
        Что уменьшаем?
      </h3>
      <button
        class="calc-ded__controls-btn calc-ded__controls-btn--active"
        type="button"
      >
        Платёж
      </button>
      <button
        class="calc-ded__controls-btn"
        type="button"
      >
        Срок
      </button>
    </div>
    <button
      class="calc-ded__btn-add"
      type="submit"
    >
      Добавить
    </button>
  </fieldset>
</DocumentFragment>
`;
